packages/next/client/use-intersection.tsx

```diff
@@ -53,6 +61,9 @@ export function useIntersection<T extends Element>({
     }
   }, [visible])
 
+  useEffect(() => {
+    if (rootEl) setMyroot(rootEl.current)
+  }, [rootEl])
```

Comment:
Hmm, on second thought we'll probably need to leave it as-is

Reply:
The weird thing here is that the 'lazyRoot' prop as a RefObject, has its "current" attribute populated only after the parent mounts and the (good or bad) thing is that when it  DOES change from 'null' and points to the element, it does not trigger re-rendering. That means that either we need to render our image only after the parent element mounts (with a usage of "useEffect" on the parent by the developer), or after the Image element mounts. So in my opinion it would be a burden for the developer to do all that so I went with the second approach. Removing the useEffect will cause the "should load the image when the lazyRoot prop is used" test to fail

Reply:
> So in my opinion it would be a burden for the developer to do all that

Agreed, thanks 👍